tutorials/io/30_reading_fnirs_data.py

```diff
@@ -61,16 +62,146 @@
 will also read the ``digaux`` data and create annotations for any triggers.
 
 
-Storing of optode locations
-===========================
+Loading legacy data in CSV or TSV format
+========================================
 
-NIRs devices consist of light sources and light detectors.
-A channel is formed by source-detector pairs.
-MNE stores the location of the channels, sources, and detectors.
-
-
-.. warning:: Information about device light wavelength is stored in
-             channel names. Manual modification of channel names is not
-             recommended.
+.. warning:: This method is not supported and users are discoraged to use it.
+             You should convert your data to the
+             `SNIRF <https://github.com/fNIRS/snirf>`_ format using the tools
+             provided by the Society for functional Near-Infrared Spectroscopy,
+             and then load it using :func:`mne.io.read_raw_snirf`.
 
+fNIRS measurements can have a non-standardised format that is not supported by
+MNE and cannot be converted easily into SNIRF. This legacy data is often in CSV
+or TSV format, we show here a way to load it even though it is not officially
+supported by MNE due to the lack of standardisation of the file format (the
+naming and ordering of channels, the type and scaling of data, and
+specification of sensor positions varies between each vendor). You will likely
+have to adapt this depending on the system from which your CSV originated.
 """  # noqa:E501
+
+import numpy as np
+import pandas as pd
+import mne
+
+# sphinx_gallery_thumbnail_number = 2
+
+###############################################################################
+# First, we generate an example CSV file which will then be loaded in to MNE.
+# This step would be skipped if you have actual data you wish to load.
+# We simulate 16 channels with 100 samples of data and save this to a file
+# called fnirs.csv.
+
+pd.DataFrame(np.random.normal(size=(16, 100))).to_csv("fnirs.csv")
```

Comment:
I wonder if it is really necessary to write out the CSV and then re-load it?  Would it be just as good to say something like "here we'll make some fake data using `pd.DataFrame`; in practice you'd use `pd.read_csv` instead to load your file"?

Reply:
I think that some users won't know how to read a csv. And this will demonstrate it.

Reply:
fair enough.